## Problem Statement:

A grocery store shared the transactional data with you. Your job is to conduct a thorough analysis of Point of Sale (POS) data, identify the most commonly occurring sets of items in the customer orders, and provide recommendations through which a grocery store can increase its revenue by popular combo offers & discounts for customers.



In [23]:
#Import the Libraries
import numpy as np
import pandas as pd

import seaborn as sns
import matplotlib.pyplot as plt

from warnings import filterwarnings
filterwarnings('ignore')


In [24]:
#Loading the file
df = pd.read_csv('dataset_group.csv', parse_dates=['Date'], dayfirst=True)

In [25]:
#Checking samples of the data
df

Date  Order_id            Product
0     2018-01-01         1             yogurt
1     2018-01-01         1               pork
2     2018-01-01         1      sandwich bags
3     2018-01-01         1         lunch meat
4     2018-01-01         1       all- purpose
...          ...       ...                ...
20636 2020-02-25      1138               soda
20637 2020-02-25      1138       paper towels
20638 2020-02-26      1139               soda
20639 2020-02-26      1139  laundry detergent
20640 2020-02-26      1139            shampoo

[20641 rows x 3 columns]

In [4]:
#Dimentions of the data
df.shape

(20641, 3)

In [5]:
#Information about the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20641 entries, 0 to 20640
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    --------------  -----         
 0   Date      20641 non-null  datetime64[ns]
 1   Order_id  20641 non-null  int64         
 2   Product   20641 non-null  object        
dtypes: datetime64[ns](1), int64(1), object(1)
memory usage: 483.9+ KB


##Exploratory Data Analysis

In [8]:
#Missing values
df.isnull().sum()

Date        0
Order_id    0
Product     0
dtype: int64

- No missing values found

In [26]:
#Checking duplicate values
print("Total duplicate values:" ,df.duplicated().sum())

Total duplicate values: 4730


In [12]:
#Let's validate if the duplicates are of same order are belong to different orders
df[df.duplicated(keep=False)]


Date  Order_id        Product
4     2018-01-01         1   all- purpose
10    2018-01-01         1   all- purpose
11    2018-01-01         1   dinner rolls
13    2018-01-01         1   all- purpose
18    2018-01-01         1   dinner rolls
...          ...       ...            ...
20632 2020-02-25      1138  sandwich bags
20633 2020-02-25      1138   toilet paper
20634 2020-02-25      1138           soda
20635 2020-02-25      1138           soda
20636 2020-02-25      1138           soda

[8613 rows x 3 columns]

- There are 4730 duplicates found in the data that are belonging to same set of orders, let's remove them

In [27]:
df = df.drop_duplicates()

In [28]:
#Descriptive Summary of the dataset - Numeric features

display(round(df.describe(),2).T)

count    mean     std  min    25%    50%    75%     max
Order_id  15911.0  574.15  328.54  1.0  289.5  579.0  859.0  1139.0

In [29]:
#Categorical
df.describe(include = 'object').T

count unique      top freq
Product  15911     37  poultry  480

In [30]:
#Checking for anomalies in the data
df['Product'].unique()

array(['yogurt', 'pork', 'sandwich bags', 'lunch meat', 'all- purpose',
       'flour', 'soda', 'butter', 'beef', 'aluminum foil', 'dinner rolls',
       'shampoo', 'mixes', 'soap', 'laundry detergent', 'ice cream',
       'toilet paper', 'hand soap', 'waffles', 'cheeses', 'milk',
       'dishwashing liquid/detergent', 'individual meals', 'cereals',
       'tortillas', 'spaghetti sauce', 'ketchup', 'sandwich loaves',
       'poultry', 'bagels', 'eggs', 'juice', 'pasta', 'paper towels',
       'coffee/tea', 'fruits', 'sugar'], dtype=object)

In [31]:
#Let's check any negative calues in the dataset
row_means = df.iloc[:, 1:].mean(axis=1)
# Count the number of rows where the mean is less than 0
count_negative_means = np.sum(row_means < 0)
print(count_negative_means)


0


- Data seems fine, Let's export for further analysis

In [32]:
df.shape

(15911, 3)

In [33]:
df.to_csv('dataset_group_processed.csv', index=False)


## Rest EDA on Tableau